core/src/main/scala/org/apache/spark/status/api/v1/ApplicationListResource.scala

```diff
@@ -53,12 +55,7 @@ private[v1] class ApplicationListResource(uiRoot: UIRoot) {
           attempt.startTime.getTime <= maxDate.timestamp
```

Comment:
This may be off topic but if your original concern was performance, then I'd note that there's no real need to evaluate `dateOk` by looking at all attempts again, if `statusOk` is `false`. Not sure it matters.